src/cursor/cur_std.c

```diff
@@ -1067,6 +1087,14 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
             if (WT_CURSOR_RECNO(cursor))
                 WT_ERR_MSG(
                   session, EINVAL, "cannot use prefix key search near for column store formats");
+            /*
+             * Prefix search near configuration can only be used for formats string or raw byte
+             * array.
+             */
+            if (!(WT_STREQ(cursor->key_format, "S") || WT_STREQ(cursor->key_format, "u") ||
+                  __check_format_fixed_string(cursor->key_format) == 0))
+                WT_ERR_MSG(session, EINVAL,
+                  "prefix key search near can only be used for format string or raw byte array");
```

Comment:
The error here can be impoved: "prefix key search near can only be used with string, fixed-length string or raw byte array format types". Same for the comment above.